### PR TITLE
Abort el-cheapo transactions if servers fail

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/5587/

### CODE REVIEW

File: etc/testing/arangod-common.conf [hunk 1]
```diff
@@ -3,6 +3,7 @@ line-number = false
 force-direct = false
 level = info
 level = replication=warn
+level = trx=debug
```
Comment (jsteemann) [inline etc/testing/arangod-common.conf]: can we remove this from the PR now?
```suggestion
```

File: arangod/Transaction/ClusterUtils.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  }
+  
+  std::vector<ServerID> failed = ci->getFailedServers();
+  transaction::Manager* mgr = transaction::ManagerFeature::manager();
+  TRI_ASSERT(mgr != nullptr);
+  
+  bool didWork = false;
+  if (ServerState::instance()->isCoordinator()) {
+    
+    // abort all transaction using a lead server
```
Comment (jsteemann) [inline arangod/Transaction/ClusterUtils.cpp]:
```suggestion
    // abort all transactions using a lead server
```

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4142/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4156/

**Comment by jsteemann** [comment]
There is an issue with blacklisting still.
With the branch, the following test in `shell_server_aql` fails:
```
    [FAILED]  tests/js/server/aql/aql-profiler-noncluster-nightly.js

      "testTraversalBlock1" failed: at assertion #83: { "i" : 3, "query" : "FOR i IN 1..@listRows\n        FOR v IN 0..@colRows OUTBOUND @root @@edgeCol RETURN v", "listRows" : 1000, "collectionRows" : 1, "expected" : [ { "type" : "SingletonNode", "calls" : 1, "items" : 1 }, { "type" : "CalculationNode", "calls" : 1, "items" : 1 }, { "type" : "EnumerateListNode", "calls" : 1, "items" : 1000 }, { "type" : "TraversalNode", "calls" : 1, "items" : 1000 }, { "type" : "ReturnNode", "calls" : 1, "items" : 1000 } ], "actual" : [ { "type" : "SingletonNode", "calls" : 1, "items" : 1 }, { "type" : "CalculationNode", "calls" : 1, "items" : 1 }, { "type" : "EnumerateListNode", "calls" : 1, "items" : 1000 }, { "type" : "TraversalNode", "calls" : 2, "items" : 1000 }, { "type" : "ReturnNode", "calls" : 2, "items" : 1000 } ] }: (2) is not equal to (1)
(Error
    at Object.assertEqual (/home/steemann/ArangoNoAsan/js/common/modules/jsunity/jsunity.js:109:19)
    at assertFuzzyNumArrayEquality (/home/steemann/ArangoNoAsan/js/server/modules/@arangodb/aql-profiler-test-helper.js:167:14)
    at Object.assertNodesItemsAndCalls (/home/steemann/ArangoNoAsan/js/server/modules/@arangodb/aql-profiler-test-helper.js:473:3)
    at Object.testTraversalBlock1 (tests/js/server/aql/aql-profiler-noncluster-nightly.js:286:22)
    at Object.run (/home/steemann/ArangoNoAsan/js/common/modules/jsunity/jsunity.js:521:23)
    at Object.Run [as run] (/home/steemann/ArangoNoAsan/js/common/modules/jsunity.js:173:24)
    at tests/js/server/aql/aql-profiler-noncluster-nightly.js:299:9
    at tests/js/server/aql/aql-profiler-noncluster-nightly.js:302:3
    at RunTest (/home/steemann/ArangoNoAsan/js/common/modules/jsunity.js:250:12)
    at source (eval at <anonymous> (unknown source), <anonymous>:4:52))
```
When reverting the blacklisting-related changes in the branch, the test passes.
So I guess we are now not blacklisting in some cases in which we should.

**Comment by graetzer** [comment]
fails for me if I run it on devel ?

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/5642/

**Pull request merged**